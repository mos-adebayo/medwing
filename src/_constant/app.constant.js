export const appConstants = {
    BASE_URL : 'http://localhost:1337',
    /*Action Types*/
    SET_ERROR : 'set_error',
    CLEAR_ERROR : 'clear_error',
    START_REQUEST : 'start_request',
    STOP_REQUEST : 'stop_request',
    ALL_LOCATIONS : 'all_locations',
    GET_LOCATIONS : 'get_locations',
    UPDATE_LOCATIONS : 'update_location',
    CREATE_LOCATIONS : 'create_location',
    APP_ERROR : 'error',

    MAP_ZOOM_LEVEL: 8,
    MAP_EXTRA_ZOOM_LEVEL: 12,
    REQUEST_ERROR: 1,
    REQUEST_SUCCESS: 2,

    /*App SDK Config*/
    TOM_TOM_KEY: 'oxmt4h5p1PFToAykPhSSHQtoOfM35VVx',
    TOM_TOM_MAP_SOURCE: 'vector',
    TOM_TOM_SEARCH_BOX_POSITION: 'topright',

};
