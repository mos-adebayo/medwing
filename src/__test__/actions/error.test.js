import configureStore from 'redux-mock-store';
import {errorActions} from "../../_actions/error.action";
import {appConstants} from "../../_constant";

const mockStore = configureStore();
const store = mockStore();

describe('select_actions', () => {
    beforeEach(() => {
        store.clearActions();
    });
});

describe('Error Action', () => {
    test('Dispatches `SET_ERROR` action and payload', () => {
        const setExpectedActionPayload = [
            {
                'error': 'error',
                'type': appConstants.SET_ERROR,
            },
        ];

       store.dispatch(errorActions.setError('error'));
       expect(store.getActions()).toEqual(setExpectedActionPayload);
    });
});

describe('Error Action Snapshot', () => {
    test('Dispatches the correct action and payload', () => {
        store.dispatch(errorActions.setError('error'));
        expect(store.getActions()).toMatchSnapshot();
    });
});
