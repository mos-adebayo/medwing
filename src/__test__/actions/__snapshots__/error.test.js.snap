// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Action Snapshot Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "error": "error",
    "type": "set_error",
  },
  Object {
    "error": "error",
    "type": "set_error",
  },
]
`;
